packages/@aws-cdk/aws-apigatewayv2-integrations/lib/http/aws.ts

```diff
@@ -0,0 +1,167 @@
+
+  /**
+   * The state machine to be executed
+   */
+  readonly stateMachine: IStateMachine;
+
+  /**
+   * The execution name
+   *
+   * @default - undefined
+   */
+  readonly name?: string;
+
+  /**
+   * The input parameters of execution
+   *
+   * @default - undefined
+   */
+  readonly input?: any;
+
+  /**
+   * The region of state machine
+   *
+   * @default - undefined
+   */
+  readonly region?: string;
+}
+
+/**
+ * The StepFunctions-StartExecution integration resource for HTTP API
+ */
+export class StepFunctionsStartExecutionIntegration extends StepFunctionsIntegration {
+
+  constructor(private readonly _scope: Construct, private readonly _props: StepFunctionsStartExecutionIntegrationProps) {
+    super(_props);
+  }
+
+  /**
+   *
+   * @internal
+   */
+  protected _integrationAction(): string {
+    return 'StartExecution';
+  }
+
+  /**
+   *
+   * @internal
+   */
+  protected _fulfillRole(credentialsRole: iam.IRole): void {
+    this._props.stateMachine.grantStartExecution(credentialsRole);
+    credentialsRole.attachInlinePolicy(
+      new iam.Policy(this._scope, 'AllowSfnSyncExec', {
+        statements: [
+          new iam.PolicyStatement({
+            actions: ['states:StartSyncExecution'],
+            effect: iam.Effect.ALLOW,
+            resources: ['*'],
+          }),
+        ],
+      }),
+    );
+
+  }
+
+  /**
+   *
+   * @internal
+   */
+  protected _buildRequestParameters(): { [key: string]: any } {
+    return {
+      StateMachineArn: this._props.stateMachine.stateMachineArn,
+      Name: this._props.name,
+      Input: this._props.input,
+      Region: this._props.region,
+    };
+  }
+}
+
+/**
+ * Step Functions StartSyncExecution integration properties
+ */
+export interface StepFunctionsStartSyncExecutionIntegrationProps extends StepFunctionsStartExecutionIntegrationProps {
+
+  /**
+   * Passes the AWS X-Ray trace header. The trace header can also be passed in the request payload.
+   *
+   * @default - undefined
+   */
+  readonly traceHeader?: string;
+}
+
+/**
+ * The StepFunctions-StartExecution integration resource for HTTP API
+ */
+export class StepFunctionsStartSyncExecutionIntegration extends StepFunctionsIntegration {
+
+  constructor(private readonly _scope: Construct, private readonly _props: StepFunctionsStartSyncExecutionIntegrationProps) {
+    super(_props);
+  }
+
+  /**
+   *
+   * @internal
+   */
+  protected _integrationAction(): string {
+    return 'StartSyncExecution';
+  }
+
+  /**
+   *
+   * @internal
+   */
+  protected _fulfillRole(credentialsRole: iam.IRole): void {
+
+    this._props.stateMachine.grantExecution(credentialsRole.grantPrincipal, 'states:StartSyncExecution');
+    credentialsRole.attachInlinePolicy(
+      new iam.Policy(this._scope, 'AllowSfnSyncExec', {
+        statements: [
+          new iam.PolicyStatement({
+            actions: ['states:StartSyncExecution'],
+            effect: iam.Effect.ALLOW,
+            resources: ['*'],
```

Comment:
I believe permission to * should be avoided as you have the state machine. It should give permission only to the state machine ARN as a best practice (least privilege).

Reply:
Yes that is correct. I will fix this. Thank you.